lib/rbac/filterer.rb

```diff
@@ -175,9 +175,6 @@ def search(options = {})
         ids_clause = ["#{klass.table_name}.id IN (?)", target_ids] if klass.respond_to?(:table_name)
       else # targets is a scope, class, or AASM class (VimPerformanceDaily in particular)
         targets = to_class(targets)
-        # could just call all on everything, but that will display deprecation warnings
-        targets = targets.all if targets < ActiveRecord::Base || targets.kind_of?(ActsAsArModel)
-
         scope = apply_scope(targets, scope)
```

Comment:
Yea this is confusing. `targets` represents the `:targets` passed in.

Both `:targets` and `:class` are basically the same thing. They can come in as a string/class name, ActiveRecord class, AAArM class, scope, array of ids, or array of objects.

The purpose of this logic is to convert all of those options into a class and scope / `klass` and `scope`

The original code was: `targets = to_class(targets).all` but we were getting deprecation messages.


My long term goal is to merge target and class, options, and reduce the sheer number of valid options.